model/labels/regexp.go

```diff
@@ -69,7 +69,7 @@ func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
 		}
 		// Simplify the syntax tree to run faster.
 		parsed = parsed.Simplify()
-		m.re, err = regexp.Compile("^(?:" + parsed.String() + ")$")
+		m.re, err = regexp.Compile("^(?s:" + parsed.String() + ")$")
```

Comment:
I think this is too late to add single-line mode.  The `parsed` version prior to this addition is used for analysis.
Suggests that something is lacking in unit tests, because I expect this change to break the logic.